### DISCUSSION TITLE
Keep-ipynb in the notebooks folder

### DISCUSSION THREAD

**Discussion by fredguth** [quarto-dev/quarto-cli#8533] [Feature Requests]
### Description

I am following the Manuscript guide with the VS Code template repo. 

I changed `_quarto.yml` to keep the intermediate .ipynb file:

```_quarto.yml
project:
  type: manuscript

execute:
  freeze: auto
  keep-ipynb: true
...
```

It creates and keeps an `index.ipynb` in the root folder. Shouldn't this file be in the notebooks folder? Maybe with the name `notebooks/Article_Notebook.ipynb` instead of `index.ipynb`.

p.s. I am not sure if this is a feature request or just a discussion I would like to have about an implementation detail.

**Comment by cscheid** [comment]
> Shouldn't this file be in the notebooks folder?

No. `keep-ipynb` is a debugging aid. It preserves the intermediate file that Quarto generates, and is placed in the same directory because it's not meant as an output artifact (it won't have any of the output features you expect, like resolved crossrefs etc). You need to use `format: ipynb` if you want that.

**Comment by fredguth** [comment]
Ah, ok. Got it.  I like the fact that .qmd are easy to track in git, but am used to write in Jupyter. As the rendered html showed the Article Notebook with other `notebooks`, I thought there would be a good place. 

I am still thinking about how should I structure my folders (I also use `nbdev`).  Is there a way to config where to keep the ipynb?

**Reply by mcanouil** [reply]
The best you can do is change the name of the notebook (`output-file`) at the format level as `output-dir` is a project wide option with no fine grained control (for now).
And if you are talking about the notebooks linked to the HTML document/website, I don't think there is a way to set its location.

**Comment by alan-isaac** [comment]
If the _ipynb intermediate files are for debugging, then perhaps they should go in a `_debugging` folder? It is unpleasant to find them  in the project folder. Even more important, they do not honor the `strip-comments` setting of the output format, but I do not want those comments to be anywhere except in my source files (`.qmd`).
